spec/app_support/reservations/rendering_spec.rb

```diff
@@ -0,0 +1,230 @@
+require "spec_helper"
+
+describe Reservations::Rendering do
+  subject(:reservation) do
+    build_stubbed(
+      :reservation,
+      reserve_start_at: reserve_start_at,
+      reserve_end_at: reserve_end_at,
+      actual_start_at: actual_start_at,
+      actual_end_at: actual_end_at,
+      canceled_at: canceled_at,
+    )
+  end
+
+  let(:reserve_start_at) { nil }
+  let(:reserve_end_at) { nil }
+  let(:actual_start_at) { nil }
+  let(:actual_end_at) { nil }
+  let(:canceled_at) { nil }
+
+  describe "#actuals_string" do
+    context "when there is an actual start_at" do
+      let(:actual_start_at) { DateTime.new(2015, 6, 1, 13, 14, 15) }
+
+      context "and an actual end_at" do
+        let(:actual_end_at) { DateTime.new(2015, 6, 1, 14, 15, 16) }
+
+        it "returns the formatted range" do
+          expect(reservation.actuals_string)
+            .to eq("Mon, 06/01/2015 8:14 AM - 9:15 AM")
+        end
+      end
+
+      context "and no actual end_at" do
+        it "formats the start time and uses '???' for the end time" do
+          expect(reservation.actuals_string)
+            .to eq("Mon, 06/01/2015 8:14 AM - ???")
+        end
+      end
+    end
+
+    context "when there is no actual start_at" do
+      context "and there is an actual end_at" do
+        let(:actual_end_at) { DateTime.new(2015, 6, 1, 14, 15, 16) }
+
+        it "uses '???' for the start time and formats the end time" do
+          expect(reservation.actuals_string)
+            .to eq("??? - Mon, 06/01/2015 9:15 AM ") # NOTE trailing space
+        end
+      end
+
+      context "and there is no actual end_at" do
+        it "returns a no actual times message" do
+          expect(reservation.actuals_string).to eq("No actual times recorded")
+        end
+      end
+    end
+  end
+
+  describe "#as_calendar_object" do
+    let(:actual_start_at) { DateTime.new(2015, 8, 1, 14, 15, 16) }
+    let(:actual_end_at) { DateTime.new(2015, 8, 1, 15, 16, 17) }
+    let(:title) { "Admin\nReservation" }
```

Comment:
Maybe not the best explanation, but I wrote the test to match what the code does. I don't know the reasoning behind it looks like it originated here: 06bb0685f5b254f8ddbd7cfd917dcd7e022fb7ac. Switching this to a space didn't seem to make a difference in how the tooltips render.